docs/readthedocs/source/doc/Nano/Overview/tensorflow_inference.md

```diff
@@ -1,98 +1,208 @@
+
+- `batch`: Specify the batch size of the dataset. This will only take effect on evaluation. If it's not set, then we use `batch=1` for evaluation.
 
 **Accuracy Control with INC**
+
 There are a few arguments required only by INC.
 - `tuning_strategy` (optional): it specifies the algorithm to search the tuning space. In most cases, you don't need to change it.
-- `timeout`: Timeout of your tuning. Defaults `0` means endless time for tuning.
+- `timeout`: Timeout of your tuning. Default: 0, means endless time for tuning.
 - `inputs`:      A list of input names. Default: None, automatically get names from the graph.
 - `outputs`:     A list of output names. Default: None, automatically get names from the graph.
+Here is an example to use INC with accuracy control as below. It will search for a model within 1% accuracy drop with 10 trials.
+
 Here is an example to use INC with accuracy control as below. It will search for a model within 1% accuracy drop with 10 trials.
 ```python
 from torchmetrics.classification import Accuracy
 
-q_model = model.quantize(precision='int8',
-                         accelerator=None,
-                         calib_dataset= train_dataset,
-                         metric=Accuracy(),
-                         accuracy_criterion={'relative': 0.01, 'higher_is_better': True},
-                         approach='static',
-                         tuning_strategy='bayesian',
-                         timeout=0,
-                         max_trials=10,
-                         )
-
-# run simple prediction
+q_model = InferenceOptimizer.quantize(model,
+                                      x=train_dataset,
+                                      accelerator=None,
+                                      metric=Accuracy(),
+                                      accuracy_criterion={'relative': 0.01,
+                                                          'higher_is_better': True},
+                                      approach='static',
+                                      tuning_strategy='bayesian',
+                                      timeout=0,
+                                      max_trials=10)
+
+# use the quantized model as before
 y_hat = q_model(train_examples)
-
-# evaluate, predict also support acceleration
-q_model.evaluate(train_dataset)
 q_model.predict(train_dataset)
-```
\ No newline at end of file
+```
+
+### Automatically Choose the Best Optimization
+
+If you have no idea about which one optimization to choose or you just want to compare them and choose the best one, you can use `InferenceOptimizer.optimize`.
+
+Still taking the example in [Runtime Acceleration](#runtime-acceleration), you can use it as following:
+
+```python
+# try all supproted optimizations
+opt = InferenceOptimizer()
+opt.optimize(model, x=train_dataset)
+
+# get the best optimization
+best_model = opt.get_best_model()
```

Comment:
`best_model, option = opt.get_best_model()`
the output of `get_best_model` is a tuple